src/opa.rs

```diff
@@ -0,0 +1,369 @@
+//! This module offers common access to the [`OpaConfig`] which can be used in operators
+//! to specify a name for a [`k8s_openapi::api::core::v1::ConfigMap`] and a package name
+//! for OPA rules.
+//!
+//! Additionally several methods are provided to build an URL to query the OPA data API.
+//!
+//! # Example
+//! ```rust
+//! use serde::{Deserialize, Serialize};
+//! use stackable_operator::kube::CustomResource;
+//! use stackable_operator::opa::OpaConfig;
+//! use stackable_operator::schemars::{self, JsonSchema};
+//!
+//! #[derive(Clone, CustomResource, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+//! #[kube(
+//!     group = "test.stackable.tech",
+//!     version = "v1alpha1",
+//!     kind = "TestCluster",
+//!     plural = "testclusters",
+//!     shortname = "test",
+//!     namespaced,
+//! )]
+//! #[serde(rename_all = "camelCase")]
+//! pub struct TestClusterSpec {
+//!     opa: Option<OpaConfig>    
+//! }
+//!
+//! let cluster: TestCluster = serde_yaml::from_str(
+//!     "
+//!     apiVersion: test.stackable.tech/v1alpha1
+//!     kind: TestCluster
+//!     metadata:
+//!       name: simple-test
+//!     spec:
+//!       opa:
+//!         configMapName: simple-opa
+//!         package: test
+//!     ",
+//!     ).unwrap();
+//!
+//! let opa_config: &OpaConfig = cluster.spec.opa.as_ref().unwrap();
+//!
+//! assert_eq!(opa_config.package_url(&cluster), "v1/data/test".to_string());
+//! assert_eq!(opa_config.full_package_url(&cluster, "http://localhost:8081"), "http://localhost:8081/v1/data/test".to_string());
+//! assert_eq!(opa_config.rule_url(&cluster, Some("myrule")), "v1/data/test/myrule".to_string());
+//! assert_eq!(opa_config.full_rule_url(&cluster, "http://localhost:8081", Some("myrule")), "http://localhost:8081/v1/data/test/myrule".to_string());
+//! ```
+use crate::client::Client;
+use crate::error;
+use crate::error::OperatorResult;
+use k8s_openapi::api::core::v1::ConfigMap;
+use kube::ResourceExt;
+use schemars::{self, JsonSchema};
+use serde::{Deserialize, Serialize};
+
+const OPA_API: &str = "v1/data";
```

Comment:
I apologize in advance for this comment, I know that I said in the call we are fine to hard code this, but if we do it here I believe we create an artificial requirement for all our authorizers to always use the same version of the data api.
I am perfectly fine to have this hard coded here as a default value (maybe with a different name like 'V1_DATA_API" or something similar that allows for easier addition of other apis later) the individual authorizers have to be able to override this. And by that I do not mean that the user should be able to configure this, but that the authorizer code should be able to do so, still hardcoded, just somewhere else :)

Reply:
introduced `OpaApiVersion` enum to differentiate the api version and respective pathes